### PR TITLE
chore(e2e-tests): track coverage

### PR DESCRIPTION
**Pull request by mzedel**
Changelog: None
Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -72,8 +73,8 @@ test:acceptance:
 
 test:acceptance:enterprise:
   extends: .template:test:acceptance
-  only:
-    - master
+  # only:
+  #   - master
```
Comment (tranchitella) [inline .gitlab-ci.yml:77]: Is this intentional?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -215,22 +216,39 @@ test:staging-deployment:webkit:
...
+    - test:unit
+  script:
+    - export COVERALLS_SERVICE_JOB_NUMBER=unittests
+    - export COVERALLS_FLAG_NAME=unittests
+    - coveralls < coverage/lcov.info
+
+publish:e2e-tests:
+  extends: .template:publish:tests
+  # only:
+  #   - master
```
Comment (tranchitella) [inline .gitlab-ci.yml:245]: Same here

File: .gitlab-ci.yml (hunk 1 above)
Comment (mzedel) [reply .gitlab-ci.yml]: nope - thanks for catching it! that was needed for the evaluation (the tests + corresponding coverage collection should still only run on master)

**Review by tranchitella — APPROVED** [review]

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -221,10 +221,10 @@ test:staging-deployment:webkit:
   before_script:
     - export CI_BUILD_REF=${CI_COMMIT_SHA}
     - export CI_BUILD_REF_NAME=${CI_COMMIT_REF_NAME}
-    - export CI_BUILD_NAME=${CI_JOB_NAME}
-    - export CI_BUILD_ID=${CI_JOB_ID}
     - export CI_MERGE_REQUEST_IID=${CI_COMMIT_BRANCH#pr_}
+    - export COVERALLS_PARALLEL=true
```
Comment (oleorhagen) [inline .gitlab-ci.yml:225]: :fist_raised:

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -221,10 +221,10 @@ test:staging-deployment:webkit:
...
     - export COVERALLS_SERVICE_JOB_ID=${CI_JOB_ID}
+    - export COVERALLS_SERVICE_NUMBER=${CI_PIPELINE_ID}
```
Comment (oleorhagen) [inline .gitlab-ci.yml:227]: :+1:

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -264,3 +266,9 @@ publish:disclaimer:
       - disclaimer.txt
   only:
     - /^(master|staging|[0-9]+\.[0-9]+\.x)$/
+
+coveralls:done:
```
Comment (oleorhagen) [inline .gitlab-ci.yml:270]: Relearning Coveralls.. Always fun :)

**Pull request merged**